### PR TITLE
merge main(394d81d) in dynamic_link branch 

### PR DESCRIPTION
**Pull request by da1suk8**
# Description

bump up dynamic_link branch
I bumped up to https://github.com/line/wasmvm/commit/394d81d63a8b7ed3215f255e49323133cf56775a

This PR is related to https://github.com/line/cosmwasm/pull/271

When https://github.com/line/cosmwasm/pull/271 is merged, update commit-hash, testdata, and build file. 

- [x] commit-hash
- [x] testdata
- [x] build file

CI is failing for the same reason as below.
https://github.com/line/wasmvm/pull/89#issuecomment-1449333630

> CI fails with a race error and this error is caused even without this PR. A test added with this PR catch the race error of the cGetContractEnv.

This issue will be resolved in https://github.com/line/cosmwasm/issues/273

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/line/wasmvm/blob/main/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

File: libwasmvm/Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [package]
 name = "wasmvm"
-version = "1.0.0-0.10.0"
+version = "1.1.1-0.11.1"
 publish = false
 authors = ["LINE Plus Corporation"]
-edition = "2018"
+edition = "2021"
```
Comment (dudong2) [inline libwasmvm/Cargo.toml:6]:
```suggestion
edition = "2023"
```

**Comment by dudong2** [comment]
In CHANGELOG.md, definitions for each version's links are added at the bottom, but `unreleased` remains, and there are no definitions for `v1.1.1-0.11.0`, `v1.1.1-0.11.1`.

File: libwasmvm/Cargo.toml (hunk 1 above)
Comment (loloicci) [reply libwasmvm/Cargo.toml:6]: next rust edition of 2021 is 2024

**Comment by loloicci** [comment]
LGTM without todo

**Comment by da1suk8** [comment]
CI is failing for the same reason as below.

https://github.com/line/wasmvm/pull/89#issuecomment-1449333630

> CI fails with a race error and this error is caused even without this PR. A test added with this PR catch the race error of the cGetContractEnv. And we will solve this error in another PR.

https://github.com/line/cosmwasm/issues/273

**Review by loloicci — APPROVED** [review]

**Review by kokeshiM0chi — APPROVED** [review]

**Pull request merged**